### PR TITLE
add mpxv model to VADR 1.4.2

### PR DESCRIPTION
**Pull request by kapsakcj**
Also updated README.md with specific versions and links to MPXV documentation. Also added an example command for MPXV assembly annotation

rendered markdown viewable here: https://github.com/StaPH-B/docker-builds/blob/cjk-vadr-mpxv/vadr/1.4.2/README.md

And FYI, running VADR on a MPXV genome can take upwards of 30 min to finish, even with 8 CPUs. More explanation on the VADR MPXV documentation.

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by kapsakcj** [comment]
How you could test this yourself locally (since 30 min runtime in GH Actions isn't a good idea):

Download NCBI `datasets` tool:

```bash
wget https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets && \
chmod +x datasets
```

Use `datasets` to download a MPXV genome from NCBI, I just picked an accession that has a high genome length that is close to the reference genome length: https://www.ncbi.nlm.nih.gov/nuccore/OP022170

You can search for more genomes here: https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?SeqType_s=Nucleotide&VirusLineage_ss=Monkeypox%20virus%20(monkeypox),%20taxid:10244

```bash
./datasets download virus genome accession OP022170.1 --filename OP022170.1.zip
unzip OP022170.1.zip

# FASTA is here:
$ ls ncbi_dataset/data/
cds.fna  data_report.jsonl  dataset_catalog.json  genomic.fna  protein.faa  virus_dataset.md
```

build the docker image:
```
# specify path to Dockerfile (must have the cjk-vadr-mpxv branch checked out)
docker build -t vadr:1.4.2 vadr/1.4.2
```

Run the docker image on the FASTA downloaded from NCBI:
```
# trim terminal Ns first
docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data vadr:1.4.2 /bin/bash -c "fasta-trim-terminal-ambigs.pl /data/ncbi_dataset/data/genomic.fna --minlen 50 --maxlen 210000 >/data/OP022170.1.trimmed.fasta"

# run v-annotate.pl on trimmed fasta
docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data vadr:1.4.2 v-annotate.pl --split --cpu 4 --glsearch -s -r --nomisc --mkey mpxv --r_lowsimok --r_lowsimxd 100 --r_lowsimxl 2000 --alt_pass discontn,dupregin OP022170.1.trimmed.fasta OP022170.1-mpxv-vadr-1.4.2-test-output
```

This is the output I received, look like this genome has fatal VADR alerts. Would be interesting to see other mpxv genomes
```
# v-annotate.pl :: classify and annotate sequences using a model library
# VADR 1.4.2 (Jul 2022)
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# date:              Thu Jul 21 16:48:27 2022
# $VADRBIOEASELDIR:  /opt/vadr/Bio-Easel
# $VADRBLASTDIR:     /opt/vadr/ncbi-blast/bin
# $VADREASELDIR:     /opt/vadr/infernal/binaries
# $VADRFASTADIR:     /opt/vadr/fasta/bin
# $VADRINFERNALDIR:  /opt/vadr/infernal/binaries
# $VADRMODELDIR:     /opt/vadr/vadr-models
# $VADRSCRIPTSDIR:   /opt/vadr/vadr
#
# sequence file:                                                                  OP022170.1.trimmed.fasta
# output directory:                                                               OP022170.1-mpxv-vadr-1.4.2-test-output
# specify that alert codes in <s> do not cause FAILure:                           discontn,dupregin [--alt_pass]
# .cm, .minfo, blastn .fa files in $VADRMODELDIR start with key <s>, not 'vadr':  mpxv [--mkey]
# in feature table for failed seqs, never change feature type to misc_feature:    yes [--nomisc]
# align with glsearch from the FASTA package, not to a cm with cmalign:           yes [--glsearch]
# use top-scoring HSP from blastn to seed the alignment:                          yes [-s]
# replace stretches of Ns with expected nts, where possible:                      yes [-r]
# with -r, do not report lowsim{5,3,i}s within identified N-rich regions:         yes [--r_lowsimok]
# w/--r_lowsimok, maximum length of N-rich region is <n>:                         2000 [--r_lowsimxl]
# w/--r_lowsimok, max diff from expected length for N-rich region is <n>:         100 [--r_lowsimxd]
# split input file into chunks, run each chunk separately:                        yes [--split]
# parallelize across <n> CPU workers (requires --split or --glsearch):            4 [--cpu]
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Validating input                                                                                          ... done. [    0.1 seconds]
# Splitting sequence file into chunks to run independently in parallel on 4 processors                      ... done. [    0.0 seconds]
# Executing 1 script to process 1 partition(s) of all 1 sequence(s)                                         ... done. [   55.0 seconds]
# Merging and finalizing output                                                                             ... done. [    0.1 seconds]
#
# Summary of classified sequences:
#
#                                                 num   num   num
#idx  model      group          subgroup         seqs  pass  fail
#---  ---------  -------------  ---------------  ----  ----  ----
1     NC_063383  Orthopoxvirus  Monkeypox_virus     1     0     1
#---  ---------  -------------  ---------------  ----  ----  ----
-     *all*      -              -                   1     0     1
-     *none*     -              -                   0     0     0
#---  ---------  -------------  ---------------  ----  ----  ----
#
# Summary of reported alerts:
#
#     alert     causes   short                            per    num   num  long
#idx  code      failure  description                     type  cases  seqs  description
#---  --------  -------  --------------------------  --------  -----  ----  -----------
1     dupregin  no       DUPLICATE_REGIONS           sequence     24     1  similarity to a model region occurs more than once
2     discontn  no       DISCONTINUOUS_SIMILARITY    sequence      1     1  not all hits are in the same order in the sequence and the homology model
3     ambgntrp  no       N_RICH_REGION_NOT_REPLACED  sequence      1     1  N-rich region of unexpected length not replaced
#---  --------  -------  --------------------------  --------  -----  ----  -----------
4     lowsim5s  yes      LOW_SIMILARITY_START        sequence      1     1  significant similarity not detected at 5' end of the sequence
5     lowsim3s  yes      LOW_SIMILARITY_END          sequence      1     1  significant similarity not detected at 3' end of the sequence
#---  --------  -------  --------------------------  --------  -----  ----  -----------
#
# Output printed to screen saved in:                              OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.log
# List of executed commands saved in:                             OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.cmd
# List and description of all output files saved in:              OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.filelist
# esl-seqstat -a output for input fasta file saved in:            OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.seqstat
# 5 column feature table output for passing sequences saved in:   OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.pass.tbl
# 5 column feature table output for failing sequences saved in:   OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.fail.tbl
# list of passing sequences saved in:                             OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.pass.list
# list of failing sequences saved in:                             OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.fail.list
# list of alerts in the feature tables saved in:                  OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.alt.list
# fasta file with passing sequences saved in:                     OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.pass.fa
# fasta file with failing sequences saved in:                     OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.fail.fa
# per-sequence tabular annotation summary file saved in:          OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.sqa
# per-sequence tabular classification summary file saved in:      OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.sqc
# per-feature tabular summary file saved in:                      OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.ftr
# per-model-segment tabular summary file saved in:                OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.sgm
# per-alert tabular summary file saved in:                        OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.alt
# alert count tabular summary file saved in:                      OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.alc
# per-model tabular summary file saved in:                        OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.mdl
# alignment doctoring tabular summary file saved in:              OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.dcr
# seed alignment summary file (-s) saved in:                      OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.sda
# replaced stretches of Ns summary file (-r) saved in:            OP022170.1-mpxv-vadr-1.4.2-test-output.vadr.rpn
#
# All output files created in directory ./OP022170.1-mpxv-vadr-1.4.2-test-output/
#
# Elapsed time:  00:00:55.13
#                hh:mm:ss
#
[ok]
```

**Comment by kapsakcj** [comment]
hmmm that only took 55 seconds 🤔 

Perhaps we should add this to the test layer

File: vadr/1.4.2/Dockerfile [hunk 1]
```diff
@@ -99,4 +104,4 @@ RUN v-annotate.pl -h && \
        "SRR13957123-vadr-outdir" && \
   ls SRR13957123-vadr-outdir
 
- 
\ No newline at end of file
+ ## no test for MPXV sequences yet, it would take too long to do via GH actions (can take up to 30min/genome)
```
Comment (erinyoung) [inline vadr/1.4.2/Dockerfile:107]: I'm not sure what to do in this case.

We should probably warn people that using vadr with MPX takes forever.

**Review by erinyoung — COMMENTED** [review]
I'm going to test this locally to make sure the MPX model is working.

File: vadr/1.4.2/Dockerfile (hunk 1 above)
Comment (kapsakcj) [reply vadr/1.4.2/Dockerfile:107]: I'm trying to find a high quality genome that would run quickly through VADR in the test layer. potentially even the one in my comment above

**Comment by kapsakcj** [comment]
OK I added a test layer for MPXV annotation. Only takes 55 seconds to run on 2 CPUs so I think it will be fine in GitHub actions. 

The genome I used does raise fatal VADR alerts, but hey I think that means that VADR is working properly.

**Comment by erinyoung** [comment]
Works on my end!

File: vadr/1.4.2/README.md [hunk 2]
```diff
@@ -4,36 +4,54 @@ Main tool : [VADR](https://github.com/ncbi/vadr)
...
 
 Full documentation: https://github.com/ncbi/vadr#vadr-documentation-
 
 Docs on Coronavirus annotation: https://github.com/ncbi/vadr/wiki/Coronavirus-annotation
 
+Docs on MPXV annotation: https://github.com/ncbi/vadr/wiki/Monkeypox-virus-annotation
+
+**Note:** Support for MonkeyPox genome annotation was just added in the VADR software (July 2022) and is under active development. Things may change quickly. See the above documentation ^ to see the latest information on the state of MPXV annotation with VADR.
+
+Also be aware that some sequences may take up to **30 minutes** to annotate, depending on how divergent it is from the RefSeq NC_063383 sequence. Some sequences may only take a minute or so.
```
Comment (erinyoung) [inline vadr/1.4.2/README.md:32]: I like the warning

**Review by erinyoung — APPROVED** [review]

**Comment by erinyoung** [comment]
I'm going to deploy this to dockerhub and quay and overwrite the latest tag.

**Comment by kapsakcj** [comment]
I'm thinking we just overwrite the existing `staphb/vadr:1.4.2` and `:latest` docker image tags. That OK with you?

Or do you want something separate like `staphb/vadr:1.4.2-mpxv`?

**Comment by kevinlibuit** [comment]
Thanks for the quick review, @erinyoung!

**Pull request merged**